### PR TITLE
New Callback: EarlyBaselineStopping

### PR DESCRIPTION
**Pull request by nittanycolonial**
Google Doc: https://docs.google.com/document/d/1FvMlmW9diNaKJ988njcXEqb862kZUpGmLWzEvla3_FA/edit?usp=sharing

### CODE REVIEW

**Comment by taehoonlee** [comment]
Thank you for the PR @nittanycolonial. Please add a unit test to `tests/keras/test_callbacks.py`.

**Comment by nittanycolonial** [comment]
@taehoonlee Unit tests have been added.

**Comment by nittanycolonial** [comment]
@taehoonlee I appreciate all the feedback. I refactored the unit tests and updated this PR.

File: keras/callbacks.py [hunk 1]
```diff
@@ -539,6 +539,81 @@ def on_train_end(self, logs=None):
             print('Epoch %05d: early stopping' % (self.stopped_epoch + 1))
 
 
+class EarlyBaselineStopping(Callback):
```
Comment (fchollet) [inline keras/callbacks.py]: This is a specific instance of early stopping. Could we instead incorporate this functionality into `EarlyStopping` (`baseline` argument in constructor), or at least leverage shared code and make this a subclass of `EarlyStopping` (i.e. by setting `self.best =baseline`)?

**Comment by fchollet** [comment]
Restarting tests.

**Comment by nittanycolonial** [comment]
@fchollet Understood. I will make it a subclass and you will see that in the next iteration

**Comment by fchollet** [comment]
Two notes:

- New arguments should be appended at the end of a signature (unless in cases where this would be semantically inconsistent) to avoid breaking people who use positional arguments.
- `baseline=None` is a better default value in this case, since metrics may be negative and 0 may be a valid metric value.

I fixed both issues in a new commit.

**Comment by nittanycolonial** [comment]
@fchollet Understood, thank you.

**Pull request merged**